substrate/primitives/version/proc-macro/src/decl_runtime_version.rs

```diff
@@ -124,8 +124,8 @@ impl ParseRuntimeVersion {
 			parse_once(&mut self.impl_version, field_value, Self::parse_num_literal)?;
 		} else if field_name == "transaction_version" {
 			parse_once(&mut self.transaction_version, field_value, Self::parse_num_literal)?;
-		} else if field_name == "state_version" {
-			parse_once(&mut self.state_version, field_value, Self::parse_num_literal_u8)?;
```

Comment:
Would be nice to still accept the old `state_version` for backwards compatibility and emit a deprecation message prompting the users to change to `system_version`.

Also, since this is a procedural macro it could be nice to allow people to specify the versions in a "nice" way instead of just having a hardcoded magic number, for example:

```
    // instead of "system_version: 0":
    state_version: StateVersion::V0,
    extrinsics_root_state_version: StateVersion::V0,

    // instead of "system_version: 1":
    state_version: StateVersion::V1,
    extrinsics_root_state_version: StateVersion::V0,

    // instead of "system_version: 2":
    state_version: StateVersion::V1,
    extrinsics_root_state_version: StateVersion::V1,
```

This would be nicer from a users' point of view, but I suppose it makes the macro even more "magic", so \*shrug\*. I'll leave it up to whatever @bkchr prefers.

Reply:
This is similar to my initial proposal but decided to not go that route since this adds another extra field to Runtimeversion. More here - https://github.com/polkadot-fellows/RFCs/pull/42#discussion_r1372588369

Reply:
I'm not talking about adding/changing the physical fields - the RFC's already accepted so that train already left the station. I'm just talking about providing potential syntax sugar in the macro which would turn the `state_version` + `extrinsics_root_state_version` into `system_version`.

A less magic alternative would be something like a `const fn system_version(state_version: StateVersion, extrinsics_root_state_version: StateVersion) -> u8` or `SystemVersion { state_version: ..., extrinsics_root_state_version: ... }.into()` helper or something like that.

(But, again, I don't have a particularly strong opinion here which way to go here. But at very least I'd like to not force additional churn on people by breaking backwards compatibility.)

Reply:
> `state_version` + `extrinsics_root_state_version` into `system_version`.

Not sure we should do this.

The general idea of still supporting `state_version` sounds like a good proposal. So, it prints some warning and internally converts it to `system_version`.

Reply:
However, I would not separate the fields.

Reply:
@vedhavyas and could you do this?